### PR TITLE
[ISTHMUS] Add aggregation function MIN/MAX support

### PR DESCRIPTION
**Pull request by weijietong**
add aggregation function min/max support, so the tpc-h Q2,Q15 could run success. Delivers #36

### CODE REVIEW

**Comment by jinfengni** [comment]
The build failed.  The change requires a change of function yaml in substrait repo.  Probably you need wait until the PR of substrait is merged.

**Comment by JamesRTaylor** [comment]
Looks like with `0.5.0` release of substrait, looks like min/max are included in substrait. If you update the commit of the substrait submodule, you should be able to get this working, @weijietong.

**Comment by JamesRTaylor** [comment]
LGTM

**Review by JamesRTaylor — APPROVED** [review]

**Review by JamesRTaylor — APPROVED** [review]

**Review by jacques-n — APPROVED** [review]
LGTM. Thanks @weijietong !

**Comment by weijietong** [comment]
Just curious, will the CI fetch the substrait sub-module's latest codes while it is re-running?  It seems build failed, since I triggered it to run success by pushing a empty commit last time.

**Comment by JamesRTaylor** [comment]
Nope, nothing automatic with substrait submodule commit since it’s not clear that we’d always want the latest. Instead, manually update this as part of your PR.

**Comment by JamesRTaylor** [comment]
I think it's best to set the substrait submodule commit to a release sha. I'd recommend setting it to `fdf7c19b58c60a13157983e00c5964625206b077` which is the `0.5.0` release: https://github.com/substrait-io/substrait/commit/fdf7c19b58c60a13157983e00c5964625206b077

**Comment by weijietong** [comment]
@JamesRTaylor  has changed the substrait module version to `0.5.0`

**Comment by weijietong** [comment]
Sorry, I can't reproduce the error. I suggest to update the workflow configure items. Submitting a PR should wait for a approval to allow the checking workflow work. It's a really unintelligible behavior.

**Comment by JamesRTaylor** [comment]
Might be due to your compiler settings. Or you may not have rebased against main which would pull in this new possible value. Do you see USER_DEFINED as a possible value? If so, you’ll need to include it as a case in the switch statement.

**Comment by jacques-n** [comment]
@weijietong, I think you're getting stuck on some kind of inconsistency with your git submodule. I would suspect that if you checked out this branch in a new git repository, you'd see this error. I've posted a separate patch that only updates submodule (as I think we should keep those changes separate from introducing new functionality as much as possible). My patch is here: #63 .

**Comment by jacques-n** [comment]
I forge pushed a rebase on top of main and removed the module change from this patch.

**Comment by weijietong** [comment]
have no idea to this problem, I git cloned a fresh repository into my local pc. Maybe due to rebase @jacques-n 's new push, I still got success on my local side.  

Anyway, I find our gradle script declared a submodle update task. It will always fetch the lastest version for the submodule. So @JamesRTaylor's expection to bind the submodule to a specific version is not possible.

So if one's pr depend on a to-be merged substrait update, that pr would be successed once the substrait's update is merged and the substrait-java pr is triggered to run again.

**Comment by jacques-n** [comment]
> have no idea to this problem, I git cloned a fresh repository into my local pc. Maybe due to rebase @jacques-n 's new push, I still got success on my local side.
> 
> Anyway, I find our gradle script declared a submodle update task. It will always fetch the lastest version for the submodule. So @JamesRTaylor's expection to bind the submodule to a specific version is not possible.
> 
> So if one's pr depend on a to-be merged substrait update, that pr would be successed once the substrait's update is merged and the substrait-java pr is triggered to run again.

The gradle operation shouldn't do that. If it does, I suggest we remove it entirely and just add an operation which checks that a version of the submodule is checked out (so people who do initial build don't have a failure).

**Review by jacques-n — APPROVED** [review]
Thanks for getting this working @weijietong !

**Pull request merged**
